root/dataframe/test_snapshotMT_dir.C

```diff
@@ -0,0 +1,154 @@
+  // we need a small autoflush setting to trigger partial merges with
+  // TFileMerger in the MT case, which tests
+  // https://github.com/root-project/root/issues/8226
+  opts.fAutoFlush = 10;
+
+  /****** non-jitted snapshot *******/
+  auto snapshot_tdf = d2.Snapshot<int, int, std::vector<float>, A>(
+      outTreeName, outFileName, {"b1", "b1_square", "b2_vector", "a"}, opts);
+
+  // Open the new file and list the branches of the tree
+  TFile f(outFileName);
+  TTree *t;
+  f.GetObject(outTreeName, t);
+  for (auto branch : *t->GetListOfBranches()) {
+    std::cout << "Branch: " << branch->GetName() << std::endl;
+  }
+  f.Close();
+
+  auto mean_b1 = snapshot_tdf->Mean("b1");
+  auto mean_a =
+      snapshot_tdf->Define("a_val", [](A &a) { return a.GetI(); }, {"a"})
+          .Mean("a_val");
+
+  std::cout << "Means:" << *mean_b1 << " " << *mean_a << std::endl;
+
+  if (*mean_b1 != *mean_a) {
+    std::cerr << "Error: the mean values of two branches which are supposed to "
+                 "be identical differ!\n";
+    return 1;
+  }
+
+  /****** jitted snapshot *******/
+  auto snapshot_jit = d2.Snapshot(outTreeName, outFileName,
+                                  {"b1", "b1_square", "b2_vector", "a"});
+
+  // Open the new file and list the branche of the trees
+  TFile jit_f(outFileName);
+  TTree *jit_t;
+  jit_f.GetObject(outTreeName, jit_t);
+  auto l = jit_t->GetListOfBranches();
+  for (auto branch : *l) {
+    std::cout << "Jitted branch: " << branch->GetName() << std::endl;
+  }
+  jit_f.Close();
+
+  auto jit_mean_b1 = snapshot_jit->Mean("b1");
+  auto jit_mean_a =
+      snapshot_jit->Define("a_val", [](A &a) { return a.GetI(); }, {"a"})
+          .Mean("a_val");
+
+  std::cout << "Jitted means:" << *jit_mean_b1 << " " << *jit_mean_a
+            << std::endl;
+
+  if (*jit_mean_b1 != *jit_mean_a || *mean_b1 != *jit_mean_b1) {
+    std::cerr << "Error: the jitted mean values of two branches which are "
+                 "supposed to be identical differ!\n";
+    return 1;
+  }
+
+  // now we exercise the regexp functionality
+  auto snapshot_jit2 = d2.Snapshot(outTreeName, outFileName, "b[1,2].*");
+
+  // Open the new file and list the branche of the trees
+  TFile jit_f2(outFileName);
+  TTree *jit_t2;
+  jit_f2.GetObject(outTreeName, jit_t2);
+  auto l2 = jit_t2->GetListOfBranches();
+  for (auto branch : *l2) {
+    std::cout << "Jitted branch: " << branch->GetName() << std::endl;
+  }
+  jit_f.Close();
+
+  return 0;
+}
+
+int runTest() {
+  auto fileName = "test_snapshot.root";
+  auto outFileName = "test_snapshot_output.root";
+  auto treeName = "myTree";
+  auto outTreeName = "myTree";
+  fill_tree(fileName, treeName);
+
+  std::cout << "---- Now with a tree in the root directory\n";
+  int ret = do_work(fileName, outFileName, treeName, outTreeName);
+
+  // now we put the tree in a directory
+  outFileName = "test_snapshot_inDirectory_output.root";
+  treeName = "myTree";
+  outTreeName = "a/myTree";
+
```

Comment:
I don't understand this part, the comment says that a tree is going to be put in a directory, but I don't see any code creating a tree or a file

Reply:
my bad, this part should be deleted as it was moved to the separate subdir test